tangelo/toolboxes/post_processing/post_selection.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module provides functions to create symmetry verification and postselection circuits."""
+from tangelo.linq import Circuit, Gate
+from tangelo.linq.helpers import measurement_basis_gates, pauli_string_to_of
+
+
+def ancilla_symmetry_circuit(circuit, sym_op):
+    """Appends to a circuit a symmetry operator and ancilla for symmetry verification.
+    For more details see arXiv:1807.10050.
+
+    Args:
+        circuit (Circuit): A quantum circuit to be equipped with symmetry verification
+        sym_op (string): The symmetry operator to be verified, in the form of a Pauli string
```

Comment:
Not fond of the string being what we expect from the user. As Alexandre said, we tend to manipulate `QubitOperator` objects, or openfermion-style (tuple) representations (which causes you to use `pauli_string_to_of`).

I'm not sure why we'd support the string case, I'd need to be convinced. I think for consistency with the rest of the package, we should not.

Reply:
I was thinking it might be more intuitive for the user to use Pauli strings. This is more like a human readable flag which would normally be used "by hand", where the user takes a circuit and decides which symmetry to input. It's much easier for a human to say "XX" than [(0, 'X'), (1, 'X')] or construct a whole QubitOperator. I can add functionality to work with other types just in case someone needs to do that though. 

Also, could add some consistency checks to make sure the number of qubits matches the input circuit.